### PR TITLE
Ensure PixelOperations.To(TPixel) uses scaling.

### LINKED ISSUE #2048: Conversion from RgbaVector Incorrect
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am running the latest version of ImageSharp
- [X] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [X] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### ImageSharp version

2.0.0

### Other ImageSharp packages and versions

none

### Environment (Operating system, version and so on)

win10 pro (Version 10.0.19043 Build 19043)

### .NET Framework version

4.7.2

### Description

I might be misunderstanding what's supposed to happen, but I think this is a bug:

```
var sourceImg = GetImage<RgbaVector>();

using (var clonedImg = sourceImg.CloneAs<HalfVector4>()) //<- should load the equivalent color values in the new HalfVector4 format?
{
    Rgba32 srcColor = new Rgba32();
    Rgba32 convertedColor = new Rgba32();

    sourceImg[0, 0].ToRgba32(ref srcColor);
    clonedImg[0, 0].ToRgba32(ref convertedColor);
                    
    Console.WriteLine($"{sourceImg[0, 0]} -> {srcColor}");
    Console.WriteLine($"{clonedImg[0, 0]} -> {convertedColor}");
}
```
Outputs:

```
RgbaVector(0.04, 0.04, 0.2, 1) -> Rgba32(10, 10, 51, 255)
HalfVector4(0.04, 0.04, 0.2, 1) -> Rgba32(132, 132, 153, 255)
```

Am I supposed to manually convert from the 32 bit `RgbaVector` values to the 16 bit `HalfVector4`?

### Steps to Reproduce

```
var sourceImg = GetImage<RgbaVector>();

using (var clonedImg = sourceImg.CloneAs<HalfVector4>()) //<- should load the equivalent color values in the new HalfVector4 format?
{
    Rgba32 srcColor = new Rgba32();
    Rgba32 convertedColor = new Rgba32();

    sourceImg[0, 0].ToRgba32(ref srcColor);
    clonedImg[0, 0].ToRgba32(ref convertedColor);
                    
    Console.WriteLine($"{sourceImg[0, 0]} -> {srcColor}");
    Console.WriteLine($"{clonedImg[0, 0]} -> {convertedColor}");
}
```

### Images

_No response_

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Fixes #2048

### CODE REVIEW

File: src/ImageSharp/Metadata/Profiles/Exif/Tags/ExifTag.String.cs [hunk 1]
```diff
@@ -94,7 +94,7 @@ public abstract partial class ExifTag
         /// <summary>
         /// Gets the MDFileUnits exif tag.
         /// </summary>
-        public static ExifTag<string> MDFileUnits => new ExifTag<string>(ExifTagValue.MDFileUnits);
+        public static ExifTag<string> MDFileUnits { get; } = new ExifTag<string>(ExifTagValue.MDFileUnits);
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Metadata/Profiles/Exif/Tags/ExifTag.String.cs:97]: Pushed this by accident but it was something I found the other day when looking at the EXIF tags.

File: tests/ImageSharp.Tests.ruleset [hunk 2]
```diff
@@ -1,6 +1,9 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<RuleSet Name="ImageSharp.Tests" ToolsVersion="16.0">
+<RuleSet Name="ImageSharp.Tests" ToolsVersion="17.0">
   <Include Path="..\shared-infrastructure\sixlabors.tests.ruleset" Action="Default" />
+  <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers">
+    <Rule Id="SA1313" Action="None" />
```
Comment (JimBobSquarePants) [inline tests/ImageSharp.Tests.ruleset:5]: Allows us to replace "dummy" with discard "_"

File: tests/ImageSharp.Tests/PixelFormats/PixelOperations/PixelOperationsTests.cs [hunk 3]
```diff
@@ -320,30 +320,51 @@ public void ToVector4(int count)
...
-            TPixel[] source = CreatePixelTestData(Count);
-            var expected = new TDestPixel[Count];
+            const int count = 2134;
+            TPixel[] source = CreatePixelTestData(count);
+            var expected = new TDestPixel[count];
 
             PixelConverterTests.ReferenceImplementations.To<TPixel, TDestPixel>(this.Configuration, source, expected);
 
-            TestOperation(source, expected, (s, d) => this.Operations.To(this.Configuration, s, d.GetSpan()));
+            TestOperation(source, expected, (s, d) => this.Operations.To(this.Configuration, s, d.GetSpan()), false);
```
Comment (JimBobSquarePants) [inline tests/ImageSharp.Tests/PixelFormats/PixelOperations/PixelOperationsTests.cs:367]: We need approximate comparison here now we've added all the types. Comparison is still very accurate.

File: tests/ImageSharp.Tests/PixelFormats/RgbaVectorTests.cs [hunk 4]
```diff
@@ -189,5 +189,21 @@ public void RgbaVector_FromGrey8()
             // assert
             Assert.Equal(expected, rgba.ToScaledVector4());
         }
+
+        [Fact]
+        public void Issue2048()
+        {
+            // https://github.com/SixLabors/ImageSharp/issues/2048
+            RgbaVector green = Color.Green.ToPixel<RgbaVector>();
```
Comment (JimBobSquarePants) [inline tests/ImageSharp.Tests/PixelFormats/RgbaVectorTests.cs:197]: This is actually covered in the `PixelOperation` tests but I like to be explicit for serious bugs.

File: tests/ImageSharp.Tests/TestUtilities/TestPixel.cs [hunk 5]
```diff
@@ -16,6 +17,11 @@ public TestPixel()
 
         public TestPixel(float red, float green, float blue, float alpha)
         {
+            Guard.MustBeBetweenOrEqualTo(red, 0F, 1F, nameof(red));
```
Comment (JimBobSquarePants) [inline tests/ImageSharp.Tests/TestUtilities/TestPixel.cs:20]: This was unbounded when it shouldn't be.

File: tests/ImageSharp.Tests/TestUtilities/TestPixel.cs [hunk 6]
```diff
@@ -33,14 +39,11 @@ public TestPixel(float red, float green, float blue, float alpha)
         public TPixel AsPixel()
         {
             var pix = default(TPixel);
-            pix.FromVector4(new System.Numerics.Vector4(this.Red, this.Green, this.Blue, this.Alpha));
+            pix.FromScaledVector4(new Vector4(this.Red, this.Green, this.Blue, this.Alpha));
```
Comment (JimBobSquarePants) [inline tests/ImageSharp.Tests/TestUtilities/TestPixel.cs:42]: Best to be explicit despite guard.

File: src/ImageSharp/PixelFormats/PixelImplementations/PixelOperations/RgbaVector.PixelOperations.cs [hunk 7]
```diff
@@ -21,7 +21,7 @@ public partial struct RgbaVector
         internal class PixelOperations : PixelOperations<RgbaVector>
         {
             private static readonly Lazy<PixelTypeInfo> LazyInfo =
-                new Lazy<PixelTypeInfo>(() => PixelTypeInfo.Create<RgbaVector>(PixelAlphaRepresentation.Unassociated), true);
+                new(() => PixelTypeInfo.Create<RgbaVector>(PixelAlphaRepresentation.Unassociated), true);
```
Comment (tocsoft) [inline src/ImageSharp/PixelFormats/PixelImplementations/PixelOperations/RgbaVector.PixelOperations.cs:24]: I have to say I am not a fan of this syntax for object creation... I find it a lot harder to read. especially in cases like this:

```c#
// in some piece of unchanged code
var list = new List<ObjType>();
.
.
.
// lots of lines later 
.
.
.
// in the middle of a PR change block, there is no context available while review what the hell this thing is.
list.Add(new(arg1, "arg2", 34)
{
    OptionalProperty = bar
});
```

not going to block any PRs etc over this just wanted to comment on the fact I feel it drastically harms readability in most cases... the only exception I can kind of agree on is property initialization where `var` is not allowed. Or possibly collection initializers where the Type of clear in the collection type.

**Review by tocsoft — APPROVED** [review]

File: src/ImageSharp/PixelFormats/PixelImplementations/PixelOperations/RgbaVector.PixelOperations.cs (hunk 7 above)
Comment (JimBobSquarePants) [reply src/ImageSharp/PixelFormats/PixelImplementations/PixelOperations/RgbaVector.PixelOperations.cs:24]: Yeah, definitely something to be aware of. I really like it in some scenarios though

**Pull request merged**